ddspipe_core/test/unittest/core/ddspipe/DdsPipeTest.cpp

```diff
@@ -286,19 +287,91 @@ TEST(DdsPipeTest, enable_disable)
+        ddspipe_configuration.init_enabled = true;
 
-/**
- * TODO
- */
-TEST(DdsPipeTest, reload)
-{
-    // TODO
+        auto discovery_database = std::make_shared<DiscoveryDatabase>();
+
+        test::DdsPipe ddspipe(
+            ddspipe_configuration,
+            discovery_database,
+            std::make_shared<FastPayloadPool>(),
+            std::make_shared<ParticipantsDatabase>(),
+            std::make_shared<eprosima::utils::SlotThreadPool>(test::N_THREADS)
+            );
+
+        types::DdsTopic topic_1;
+        topic_1.m_topic_name = "topic1";
+        topic_1.type_name = "type1";
+        eprosima::utils::Heritable<types::DistributedTopic> htopic_1 =
+                eprosima::utils::Heritable<types::DdsTopic>::make_heritable(topic_1);
+
+        ASSERT_FALSE(ddspipe.is_topic_discovered(htopic_1));
+        ASSERT_FALSE(ddspipe.is_topic_active(htopic_1));
+        ASSERT_FALSE(ddspipe.is_bridge_created(htopic_1));
+
+        types::Endpoint endpoint_1;
+        endpoint_1.kind = types::EndpointKind::reader;
+        endpoint_1.topic = topic_1;
+
+        discovery_database->add_endpoint(endpoint_1);
+
+        // Wait a bit for callback to arrive
+        eprosima::utils::sleep_for(10u);
+
+        ASSERT_TRUE(ddspipe.is_topic_discovered(htopic_1));
+        ASSERT_TRUE(ddspipe.is_topic_active(htopic_1));
+        ASSERT_TRUE(ddspipe.is_bridge_created(htopic_1));
+
+        // Create two Wildcard Topics
+        types::WildcardDdsFilterTopic topic_2;
+        topic_2.topic_name.set_value("topic*");
+
+        eprosima::utils::Heritable<types::IFilterTopic> htopic_2 =
+                eprosima::utils::Heritable<types::WildcardDdsFilterTopic>::make_heritable(topic_2);
+
+        types::WildcardDdsFilterTopic topic_3;
+        topic_3.topic_name.set_value("top*");
+
+        eprosima::utils::Heritable<types::IFilterTopic> htopic_3 =
+                eprosima::utils::Heritable<types::WildcardDdsFilterTopic>::make_heritable(topic_3);
+
+        // Add the Topic to the allowlist
+        DdsPipeConfiguration new_ddspipe_configuration;
+        new_ddspipe_configuration.allowlist.insert(htopic_2);
+        ddspipe.reload_configuration(new_ddspipe_configuration);
+
+        ASSERT_TRUE(ddspipe.is_topic_discovered(htopic_1));
+        ASSERT_TRUE(ddspipe.is_topic_active(htopic_1));
+        ASSERT_TRUE(ddspipe.is_bridge_created(htopic_1));
+
+        // Add the Topic to the blocklist
+        new_ddspipe_configuration.blocklist.insert(htopic_3);
+        ddspipe.reload_configuration(new_ddspipe_configuration);
+
+        ASSERT_TRUE(ddspipe.is_topic_discovered(htopic_1));
+        ASSERT_FALSE(ddspipe.is_topic_active(htopic_1));
+        ASSERT_TRUE(ddspipe.is_bridge_created(htopic_1));
+
+        // Remove the Topic from the blocklist
+        new_ddspipe_configuration.blocklist.erase(htopic_3);
+        ddspipe.reload_configuration(new_ddspipe_configuration);
+
+        ASSERT_TRUE(ddspipe.is_topic_discovered(htopic_1));
```

Comment:
Would the bridge have been created if the blocklist contained the topic at pipe's startup? In that case, that condition should be tested.